// @launchpadlab/lp-components/lib/tables/components/table-header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _utils = require('../../utils');

var _helpers = require('../helpers');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  column: _helpers.Types.column.isRequired,
  sortPath: _propTypes2.default.string,
  ascending: _propTypes2.default.bool.isRequired,
  onClick: _propTypes2.default.func
};

function TableHeader(_ref) {
  var _ref$column = _ref.column,
      name = _ref$column.name,
      label = _ref$column.label,
      disabled = _ref$column.disabled,
      sortPath = _ref.sortPath,
      ascending = _ref.ascending,
      onClick = _ref.onClick;

  var active = sortPath === name;
  var arrowClass = getArrowClass(active, ascending);
  return _react2.default.createElement(
    'th',
    {
      onClick: onClick,
      className: (0, _classnames2.default)(arrowClass, { 'sortable': !disabled })
    },
    label || (0, _utils.startCase)(name)
  );
}

function getArrowClass(active, ascending) {
  if (!active) return '';
  return ascending ? 'order-ascend' : 'order-descend';
}

TableHeader.propTypes = propTypes;

exports.default = TableHeader;
