// @launchpadlab/lp-components/lib/tables/components/table-row.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _utils = require('../../utils');

var _helpers = require('../helpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  columns: _propTypes2.default.arrayOf(_helpers.Types.column).isRequired,
  rowComponent: _helpers.Types.component,
  rowData: _propTypes2.default.any
};

var DefaultRowComponent = function DefaultRowComponent(_ref) {
  var children = _ref.children;
  return _react2.default.createElement(
    'tr',
    null,
    children
  );
}; // eslint-disable-line
var DefaultCellComponent = function DefaultCellComponent(_ref2) {
  var className = _ref2.className,
      onClick = _ref2.onClick,
      placeholder = _ref2.placeholder,
      value = _ref2.value;
  return (// eslint-disable-line
    _react2.default.createElement(
      'td',
      { className: className, onClick: onClick },
      (0, _utils.isNil)(value) ? placeholder : value
    )
  );
};

function TableRow(_ref3) {
  var columns = _ref3.columns,
      _ref3$rowComponent = _ref3.rowComponent,
      RowComponent = _ref3$rowComponent === undefined ? DefaultRowComponent : _ref3$rowComponent,
      rowData = _ref3.rowData;

  return _react2.default.createElement(
    RowComponent,
    { data: rowData },
    columns.map(function (column, key) {
      var name = column.name,
          _column$component = column.component,
          CellComponent = _column$component === undefined ? DefaultCellComponent : _column$component,
          _column$format = column.format,
          format = _column$format === undefined ? _utils.identity : _column$format,
          _column$onClick = column.onClick,
          onClick = _column$onClick === undefined ? _utils.noop : _column$onClick,
          rest = _objectWithoutProperties(column, ['name', 'component', 'format', 'onClick']);

      var value = format((0, _utils.get)(name, rowData));
      var onColClick = column.disabled ? _utils.noop : function () {
        return onClick(rowData);
      };
      return _react2.default.createElement(CellComponent, _extends({ // eslint-disable-line
        key: key,
        value: value,
        name: name,
        data: rowData,
        onClick: onColClick
      }, rest));
    })
  );
}

TableRow.propTypes = propTypes;

exports.default = TableRow;
